sensing/pointcloud_preprocessor/include/pointcloud_preprocessor/time_synchronizer/time_synchronizer_nodelet.hpp

```diff
@@ -0,0 +1,182 @@
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id: TIME_SYNCHRONIZER.cpp 35231 2011-01-14 05:33:20Z rusu $
+ *
+ */
+
+#ifndef POINTCLOUD_PREPROCESSOR__TIME_SYNCHRONIZER__TIME_SYNCHRONIZER_NODELET_HPP_
+#define POINTCLOUD_PREPROCESSOR__TIME_SYNCHRONIZER__TIME_SYNCHRONIZER_NODELET_HPP_
+
+#include <deque>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <string>
+#include <vector>
+
+// ROS includes
+#include "autoware_point_types/types.hpp"
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <point_cloud_msg_wrapper/point_cloud_msg_wrapper.hpp>
+#include <tier4_autoware_utils/ros/debug_publisher.hpp>
+#include <tier4_autoware_utils/system/stop_watch.hpp>
+
+#include <autoware_auto_vehicle_msgs/msg/velocity_report.hpp>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+#include <tier4_debug_msgs/msg/int32_stamped.hpp>
+#include <tier4_debug_msgs/msg/string_stamped.hpp>
+
+#include <message_filters/pass_through.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/sync_policies/exact_time.h>
+#include <message_filters/synchronizer.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+namespace pointcloud_preprocessor
+{
+using autoware_point_types::PointXYZI;
+using point_cloud_msg_wrapper::PointCloud2Modifier;
+/** \brief @b PointCloudDataSynchronizerComponent is a special form of data
+ * synchronizer: it listens for a set of input PointCloud messages on the same topic,
+ * checks their timestamps, and concatenates their fields together into a single
+ * PointCloud output message.
+ * \author Radu Bogdan Rusu
+ * \edited by Yoshi Ri
+ */
+class PointCloudDataSynchronizerComponent : public rclcpp::Node
+{
+public:
+  typedef sensor_msgs::msg::PointCloud2 PointCloud2;
+
+  /** \brief constructor. */
+  explicit PointCloudDataSynchronizerComponent(const rclcpp::NodeOptions & node_options);
+
+  /** \brief constructor.
+   * \param queue_size the maximum queue size
+   */
+  PointCloudDataSynchronizerComponent(const rclcpp::NodeOptions & node_options, int queue_size);
+
+  /** \brief Empty destructor. */
+  virtual ~PointCloudDataSynchronizerComponent() {}
+
+private:
+  /** \brief Delay Compensated PointCloud publisher*/
+  std::map<std::string, rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr>
+    transformed_raw_pc_publisher_map_;
+
+  /** \brief The maximum number of messages that we can store in the queue. */
+  int maximum_queue_size_ = 3;
+
+  double timeout_sec_ = 0.1;
+
+  std::set<std::string> not_subscribed_topic_names_;
+
+  /** \brief A vector of subscriber. */
+  std::vector<rclcpp::Subscription<PointCloud2>::SharedPtr> filters_;
+
+  rclcpp::Subscription<autoware_auto_vehicle_msgs::msg::VelocityReport>::SharedPtr sub_twist_;
```

Comment:
could you change velocity_report to twist_with_covariance? which is output of [vehicle_velocity_converter](https://github.com/autowarefoundation/autoware.universe/tree/main/sensing/vehicle_velocity_converter)

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/3877